rpcperms/interceptor.go

```diff
@@ -7,13 +7,53 @@ import (
 
 	"github.com/btcsuite/btclog"
 	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/lightningnetwork/lnd/lnrpc"
 	"github.com/lightningnetwork/lnd/macaroons"
 	"github.com/lightningnetwork/lnd/monitoring"
 	"google.golang.org/grpc"
 	"gopkg.in/macaroon-bakery.v2/bakery"
 )
 
+// rpcState is an enum that we use to keep track of the current RPC service
+// state. This will transition as we go from startup to unlocking the wallet,
+// and finally fully active.
+type rpcState uint8
+
+const (
+	// walletNotCreated is the starting state if the RPC server is active,
+	// but the wallet is not yet created. In this state we'll only allow
+	// calls to the WalletUnlockerService.
+	walletNotCreated rpcState = iota
+
+	// walletLocked indicates the RPC server is active, but the wallet is
+	// locked. In this state we'll only allow calls to the
+	// WalletUnlockerService.
+	walletLocked
+
+	// walletUnlocked means that the wallet has been unlocked, but the full
+	// RPC server is not yeat ready.
+	walletUnlocked
+
+	// rpcActive means that the RPC server is ready to accept calls.
+	rpcActive
+)
+
 var (
+	// ErrNoWallet is returned if the wallet does not exist.
+	ErrNoWallet = fmt.Errorf("wallet not created")
```

Comment:
Nit: Some error message suggestions:
* "wallet not created, create one to enable full RPC access"
* "wallet locked,  unlock it to enable full RPC access"
* "wallet already unlocked, WalletUnlocker service is no longer available"